### DISCUSSION TITLE
Dynamically generating pages using GraphQL queries

### DISCUSSION THREAD

**Discussion by johnletey** [vercel/next.js#11528] [Help] [answered]
The title of the discussion pretty much sums up what I want to do. But let me go into some more detail:

With Gatsby, inside your `gatsby-node.js` file you can automatically create pages using a page template and pass data into the template from a GraphQL query. I have had a look around and I can't seem to find any way to do this in Next.js.

/cc @nikitavoloboev

**Comment by timneutkens** [comment, marked as answer]
In general would recommend reading the full document here: https://nextjs.org/docs/basic-features/data-fetching

But what you're looking for specifically is `getStaticPaths`: https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation which allows you to generate dynamic paths

**Reply by timneutkens** [reply]
One example with graphql would be https://github.com/zeit/next.js/blob/canary/examples/cms-datocms/pages/posts/%5Bslug%5D.js#L53-L75

**Comment by johnletey** [comment]
Thanks for the reply @timneutkens 🙏 This is exactly what I was looking for :)
